### PR TITLE
Add : Visual Testing Handbook Chapter 1

### PR DESCRIPTION
**Pull request by gollumnima**
- [x] 어체, 오타 및 통일된 키워드로 변경 완료

### CODE REVIEW

File: content/visual-testing-handbook/react/ko/introduction.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+![최소화한 컴포넌트 코드](/tutorials/visual-testing-handbook/code-visual-testing-optimized.png)
+
+실제로 DOM 스냅샷은 HTML blob을 평가하여 UI가 렌더링되는 방식을 결정하기 어렵기 때문에 불편하다.
+
+스냅샷 테스트는 다른 자동화된 UI 테스트와 마찬가지로 취약하다. 컴포넌트의 내부 작업을 변경하려면 컴포넌트의 렌더링된 출력값이 변경되었는지 여부에 관계없이 테스트를 업데이트해야 한다.
+
+## UI를 위해 만들어진 시각적 테스팅
+시각적 테스트는 UI 모양의 변경 사항을 포착하도록 설계되었다. Storybook과 같은 컴포넌트 탐색기를 사용하여 UI 컴포넌트를 격리하고, 다양한 경우의 수를 모방하고, 지원되는 테스트 케이스를 "story"로 저장한다.
+
+개발하는 동안 브라우저에서 컴포넌트를 렌더링하여 컴포넌트가 어떻게 보이는지 빠르게 수동 확인을 위해 "실행"해라. 컴포넌트 탐색기에 나열된 각 테스트 케이스를 작동시키면서 컴포넌트의 변경사항을 확인해라.
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/introduction.md:44]: 명령어로 쓰인 부분에 대해 어떻게 번역하면 좋을지 얘기해보는 게 좋을 것 같아요!

File: content/visual-testing-handbook/react/ko/component-explorer.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+컴포넌트와 상태가 많아질수록, 사용자의 디바이스와 브라우저에서 올바르게 렌더링되는지 확인하는 건 더욱 어려워집니다.
+
+컴포넌트 탐색기는 컴포넌트에 기반하는 변경사항을 화면에 표시해 일관성을 해치는 일을 방지합니다. 이를 통해 개발자는 각각의 state와 관계없이 집중할 수 있습니다. 개별적으로 테스트를 진행할 수 있으며, 모의(mocking)를 통해 복잡한 edge case를 복제할 수 있습니다.
+
+![컴포넌트 테스트 케이스](/visual-testing-handbook/component-test-cases.png)
+
+### 빨라진 개발
+
+애플리케이션 개발은 끝나지 않습니다. 계속해서 반복해야 합니다. 그렇기 때문에 UI 아키텍처는 새로운 기능을 수용할 수 있어야 합니다. 컴포넌트 모델은 UI를 애플리케이션 비즈니스 로직과 백앤드에서 분리해 호환성이 좋습니다.
```
Comment (twinstae) [inline content/visual-testing-handbook/react/ko/component-explorer.md:36]: 1. never는 어감이 강하니까 단순 부정보다는 "절대 끝나지 않습니다"처럼 절대나 결코 를 붙여주시면 자연스러울 것 같아요.

2. adaptable의 의미까지 살리자면 "컴포넌트는 새 기능을 수용하기 위해 적응적이어야 합니다."

File: content/visual-testing-handbook/react/ko/component-explorer.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+컴포넌트 탐색기는 sandbox를 제공하고 애플리케이션과 분리된 상황에서 UI를 개발함으로써 구분을 명확하게 합니다. 즉 애플리케이션의 다른 부분에서 방해를 받거나 state가 오염되는 일 없이 각자 다른 UI를 동시에 작업할 수 있습니다.
```
Comment (twinstae) [inline content/visual-testing-handbook/react/ko/component-explorer.md:38]: by providing a sandbox 까지 번역하자면 이렇게 될 수 있지 않을까요.

"컴포넌트 탐색기는 애플리케이션에서 떨어져 독립적으로 UI를 개발할 수 있는 샌드박스를 제공해서 이런 구분을 명확하게 합니다."

File: content/visual-testing-handbook/react/ko/component-explorer.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+### 빨라진 개발
+
+애플리케이션 개발은 끝나지 않습니다. 계속해서 반복해야 합니다. 그렇기 때문에 UI 아키텍처는 새로운 기능을 수용할 수 있어야 합니다. 컴포넌트 모델은 UI를 애플리케이션 비즈니스 로직과 백앤드에서 분리해 호환성이 좋습니다.
+
+컴포넌트 탐색기는 sandbox를 제공하고 애플리케이션과 분리된 상황에서 UI를 개발함으로써 구분을 명확하게 합니다. 즉 애플리케이션의 다른 부분에서 방해를 받거나 state가 오염되는 일 없이 각자 다른 UI를 동시에 작업할 수 있습니다. 
+
+### 더 쉬워진 협업
+
+UI의 본질은 시각적이라는 것 입니다. Code로만 이루어진 pull request는 불완전한 작업의 대표주자 입니다. 진정한 협업을 하고싶다면 이해관계자가 UI를 살펴보아야 합니다.
```
Comment (twinstae) [inline content/visual-testing-handbook/react/ko/component-explorer.md:42]: 원문을 그대로 살리는 게 더 자연스럽게 느껴지네요. "~이라는 것입니다."는 군더더기 같아요.
"UI는 본질적으로 시각적입니다."

대표주자라는 말은 없어서. 원문을 그대로 살리자면...
"Code-only pull requests are an incomplete representation of the work."
"코드만으로 된 pull request는 작업을 나타내기에 불완전합니다."

File: content/visual-testing-handbook/react/ko/component-explorer.md [hunk 5]
```diff
@@ -5,15 +5,11 @@ description: 'UI 개발과 시각적 테스팅을 위한 도구'
 commit: ''
 ---
 
-현대의 UI는 셀 수 없이 많은 상태, 언어, 디바이스, 브라우저 그리고 사용자 데이터를 지원합니다. 과거에, UI를 개발하는 일은 다소 번거로운 일이었습니다. 적합한 설정값과 디바이스를 이용해서 주어진 페이지를 다뤄야만 했습니다. 그런 다음 페이지에 도달하기 위해 주변을 클릭하고 정확한 상태값을 가져야만 코딩을 시작할 수 있었습니다.
+현대의 UI는 셀 수 없이 많은 상태, 언어, 디바이스, 브라우저 그리고 사용자 데이터의 조합을 지원합니다. 과거에, UI를 개발하는 일은 다소 번거로웠습니다. 올바른 디바이스에서 적합하게 설정을 하고 주어진 페이지까지 이동해야 했습니다. 그런 다음 페이지에 도달하기 위해 이곳저곳을 클릭하면서 페이지를 올바른 상태로 옮겨두고 나서야 코딩을 시작할 수 있었습니다.
 
-You build UI components in isolation to focus on each component's supported variations. That allows you to gauge how inputs (props, state) affect the rendered UI and forms the basis of your visual test suite.
+**컴포넌트 탐색기는 비즈니스 로직과 애플리케이션 context로부터 UI문제를 분리합니다.** 컴포넌트가 지원하는 변형들에 초점을 맞추기 위해 UI 컴포넌트를 별도로 빌드합니다. 이렇게 함으로써 입력값(props, state 등)이 렌더링된 UI에 어떻게 영향을 주는지 측정하게 해주고, 시각적 test suit의 기반을 형성해줍니다.
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/component-explorer.md:10]: test suit를 한글로 번역하기 어려울까요?

File: content/visual-testing-handbook/react/ko/workflow.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+테스트 주도 방법론(TDD)를 통해 사용자는 정확한 입력값의 측면에서 어떤 코드가 필요한지 명확하게 파악할 수 있게 됩니다. (컴포넌트의 경우, 이를 "상태"라고 합니다.) 이러한 방식으로, 모듈의 모든 사례를 다루게 됩니다.
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/visual-testing-handbook/test-driven-development.mp4"
+    type="video/mp4">
+</video>
+
+예시를 하나 살펴봅시다. 원시 날짜 객체를 "2주 전"과 같은 형식으로 변환하는 `상대화(relativize)` 함수가 있다고 생각해봅시다. 다루고자 하는 모든 다양한 유형의 입력값을 간단히 설명해 보는 것은 어려운일이 아닙니다. 그런 다음, 해결책에 가까워졌다고 생각이 들때마다 "테스트(test)" 버튼을 누르세요.
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/workflow.md:24]: 어려운일 -> 어려운 일, 들때마다 -> 들 때마다

File: content/visual-testing-handbook/react/ko/workflow.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+하지만, UI 개발을 할 때는 테스트 주도 방법론(TDD)이 잘 맞진 않습니다. 테스트를 미리 정의하기 어렵고, 모듈을 분리시키는 것이 어려운 일이고, 산출값이 주관적이기 때문입니다. 이러한 단점은 컴포넌트를 개별적으로 분리해 시각적 테스트를 함으로써 해결할 수 있습니다.
+
+## 시각적 테스팅
+
+UI 테스트의 가장 어려운 부분은 코드만으론 적절한 시각적인 세부사항을 확인할 수 없다는 점입니다. 시각적 테스팅은 빠르고 집중적인 방식으로 인간의 판단을 포함한 이러한 확인절차를 건너뜁니다.
+
+#### 시각적 테스트의 작업흐름(workflow)
+
+실제로, 시각적 테스트는 Storybook을 사용해 정의된 테스트의 상태의 전체에 걸쳐 컴포넌트를 "시각적인" 방법으로 테스트 합니다.시각적 테스트는 다른 유형의 테스트 방식과 동일한 설정, 실행과 분해 단계를 공유하지만, 확인단계는 사용자의 몫입니다.
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/workflow.md:36]: 합니다.시각적 -> 합니다. 시각적

File: content/visual-testing-handbook/react/ko/workflow.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+스토리북에서 `Task`와 변경사항을 사이드바에서 확인할 수 있습니다. 이는 테스트 주기에서 _“실행단계(execute)”_ 에 해당합니다. 스토리북에서 우리가 눈으로 확인하는 것은 _“확인단계(verify)”_ 에 해당합니다.
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/visual-testing-handbook/task-stories-snoozed-optimized.mp4"
+    type="video/mp4"/>
+</video>
+
+UI 테스팅을 하면서, 육안으로 확인하는 것은 실용적인 접근입니다. 시각적인 모양에 영향을 주지 않으면서 컴포넌트를 변화시키는 코드를 작성하는 것은 견실한 방법이기 때문입니다. 또한, 미리 입려값을 작성하고 시각적으로 출력값을 확인하기만 하면 되므로, 자연스럽게 테스트 주도 방법론(TDD) 식으로 UI를 설계하게 됩니다.
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/workflow.md:115]: 입려값 -> 입력값

File: content/visual-testing-handbook/react/ko/workflow.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+  <source
+    src="/visual-testing-handbook/test-driven-development.mp4"
+    type="video/mp4">
+</video>
+
+예시를 하나 살펴봅시다. 원시 날짜 객체를 "2주 전"과 같은 형식으로 변환하는 `상대화(relativize)` 함수가 있다고 생각해봅시다. 다루고자 하는 모든 다양한 유형의 입력값을 간단히 설명해 보는 것은 어려운일이 아닙니다. 그런 다음, 해결책에 가까워졌다고 생각이 들때마다 "테스트(test)" 버튼을 누르세요.
+
+테스트 프레임워크(framework)를 사용하면 단순히 한 부분을 테스트 하기 위해 전체 애플리케이션에 입력값을 넣을 필요 없이 `상대화(relativize)` 함수를 분리해서 실행할 수 있습니다.
+
+하지만, UI 개발을 할 때는 테스트 주도 방법론(TDD)이 잘 맞진 않습니다. 테스트를 미리 정의하기 어렵고, 모듈을 분리시키는 것이 어려운 일이고, 산출값이 주관적이기 때문입니다. 이러한 단점은 컴포넌트를 개별적으로 분리해 시각적 테스트를 함으로써 해결할 수 있습니다.
```
Comment (perfumelim) [inline content/visual-testing-handbook/react/ko/workflow.md:28]: 혹시 "테스트를 미리 정의하기 어렵고, 모듈을 분리하기 어려우며, 산출값이 주관적이기 때문입니다." 로 줄이는 편이 더 문장이 자연스럽지 않을까요?  별 차이 없는 거 같긴 하지만요 ㅎ_ㅎ

File: content/visual-testing-handbook/react/ko/workflow.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+잘 짜여진 디자인대로 앱을 설계하게 되면, 디자인 산출물에 내포된 입력값과 출력값을 가진 명시적인 컴포넌트를 만들어낼 수 있습니다. 시각적 테스팅 과정을 “디자인 스펙”과 연결해보면 테스트 주도 방법론(TDD)과 상당히 유사한 방식으로 테스트를 진행할 수 있습니다.
+
+다음 장에서는, 시각적 테스트 주도 방법론(Visual TDD)을 이용해 예시 컴포넌트를 만들어보면서 지금까지 배웠던 내용을 적용해볼 것입니다.
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/visual-testing-handbook/visual-test-driven-development.mp4"
+    type="video/mp4">
+</video>
```
Comment (pak-fuse) [inline content/visual-testing-handbook/react/ko/workflow.md:127]: '스토리북'이라고 되어있는 부분은 'Storybook'으로 변환하시면 될 것 같아요 ~

File: content/visual-testing-handbook/react/ko/workflow.md (hunk 8 above)
Comment (perfumelim) [inline content/visual-testing-handbook/react/ko/workflow.md:115]: 입려값 오타인 것 같아요!
